src/mainloop.rkt

```diff
@@ -300,23 +300,23 @@
                     (list (*context*))))
   (cons domain (apply mk-pcontext pts+exs)))
 
-(define (initialize-alt-table! prog pcontext ctx)
-  (define alt (make-alt prog))
-  (*start-prog* prog)
-  (define table (make-alt-table (*pcontext*) alt ctx))
-
-  ; Add starting alt in every precision
+(define (initialize-alt-table! egraph initial initial-id pcontext context
+                               iterations-length)
+  (define alternative (make-alt initial))
+  (define table (make-alt-table pcontext alternative context))
   (^table^
-   (if (*pareto-mode*)
-       (let ([new-alts (starting-alts alt ctx)])
-         (define-values (errss costs) (atab-eval-altns table new-alts ctx))
-         (atab-add-altns table new-alts errss costs))
-       table))
-
-  (when (flag-set? 'setup 'simplify)
-      (reconstruct! (patch-table-run-simplify (atab-active-alts (^table^))))
-      (finalize-iter!)
-      (^next-alts^ #f)))
+   (if (flag-set? 'setup 'simplify)
+       (let ([simplified 
+              (remove-duplicates
+               (for/list ([iteration (in-range iterations-length)])
+                 (alt
+                  (egraph-get-simplest egraph initial-id iteration)
+                  ;; TODO: What to put where egg-query was
+                  (list 'simplify null 'todo-egg-query #f #f)
+                  (list alternative))))])
+         (define-values (errss costs) (atab-eval-altns table simplified context))
+         (atab-add-altns table simplified errss costs))
+       table)))
```

Comment:
Ok, so I don't think the right design is to move the egraph out into `mainloop.rkt`.